dropwizard/src/main/java/com/linecorp/armeria/dropwizard/ArmeriaSettings.java

```diff
@@ -0,0 +1,972 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+/*
+ * Copyright 2012-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linecorp.armeria.dropwizard;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+import javax.annotation.Nullable;
+
+import com.google.common.collect.ImmutableList;
+
+import com.linecorp.armeria.common.SessionProtocol;
+import com.linecorp.armeria.server.Server;
+import com.linecorp.armeria.server.VirtualHostBuilder;
+
+import io.netty.handler.ssl.ClientAuth;
+import io.netty.handler.ssl.SslProvider;
+
+/**
+ * Settings for armeria servers, e.g.,
+ * <pre>{@code
+ * metricsPath: "/internal/metrics"
+ * gracefulShutdownQuietPeriodMillis: 5000
+ * gracefulShutdownTimeoutMillis: 40000
+ * maxRequestLength: 10485760
+ * maxNumConnections: 5000
+ * dateHeaderEnabled: false
+ * serverHeaderEnabled: true
+ * verboseResponses: false
+ * defaultHostname: "host.name.com"
+ * ports:
+ *   - port: 8080
+ *     protocol: HTTP
+ *   - ip: 127.0.0.1
+ *     port: 8081
+ *     protocol: HTTP
+ *   - port: 8443
+ *     protocols:
+ *       - HTTPS
+ *       - PROXY
+ * ssl:
+ *   keyAlias: "host.name.com"
+ *   keyStore: "keystore.jks"
+ *   keyStorePassword: "changeme"
+ *   trustStore: "truststore.jks"
+ *   trustStorePassword: "changeme"
+ * compression:
+ *   enabled: true
+ *   mimeTypes:
+ *     - text/*
+ *     - application/json
+ *   excludedUserAgents:
+ *     - some-user-agent
+ *     - another-user-agent
+ *   minResponseSize: 1KB
```

Comment:
I wonder if we need this feature or not because this is probably not part of server settings. Thoughts?

Reply:
Not sure I understand the feature myself. Would a smaller response end up throwing a 500 error?

Reply:
Ah, it basically tells the server not to compress the content smaller than 1KB. This is not part of core Armeria but it's from Spring Boot. Maybe we should drop this for Dropwizard if Dropwizard does not have this sort of configuration out of the box?

Reply:
> This is not part of core Armeria but it's from Spring Boot.

I didn't know that. Thanks for sharing.

> Maybe we should drop this for Dropwizard if Dropwizard does not have this sort of configuration out of the box?

After your review, I found [GZip compression documentation](http://dropwizard.io/en/stable/manual/configuration.html#man-configuration-gzip) in the Dropwizard site. I think the GZip feature is similar to ArmeriaSettings' compression.

Reply:
Ah, OK. Then perhaps we can keep this section. :smile: